include/nanobind/eigen/sparse.h

```diff
@@ -0,0 +1,124 @@
+/*
+    nanobind/eigen/dense.h: type casters for dense Eigen
+    vectors and matrices
+
+    Copyright (c) 2023 Wenzel Jakob
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/tensor.h>
+#include <nanobind/eigen/dense.h>
+#include <Eigen/SparseCore>
+
+static_assert(EIGEN_VERSION_AT_LEAST(3, 3, 0),
+              "Eigen matrix support in nanobind requires Eigen >= 3.3.0");
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+
+NAMESPACE_BEGIN(detail)
+
+template <typename Scalar, int Flags, typename StorageIndex>
+using EigenMapSparseMatrix = Eigen::Map<Eigen::SparseMatrix<Scalar, Flags, StorageIndex>>;
+
+/// Caster for Eigen::SparseMatrix<T>
+template <typename T> struct type_caster<T, enable_if_t<is_eigen_sparse_v<T>>> {
+    using Scalar = typename T::Scalar;
+    using StorageIndex = typename T::StorageIndex;
+    using Index = typename T::Index;
+    static constexpr bool row_major = T::IsRowMajor;
+
+    using ScalarTensor = tensor<Scalar, shape<any>, c_contig, device::cpu>;
```

Comment:
I think one crucial thing is still missing in these type annotations: `numpy`! Otherwise, nanobind will call the SciPy sparse matrix constructor with a bunch of DLPack capsules, and SciPy does not know what to do with hose.

Reply:
Thanks! That's it.